Include support for recent GCC versions in config and comp scripts.

Written by Oliver Stueker, ACENET/Compute Canada
Memorial University of Newfoundland, St. John's, Canada
--- config.orig	2017-07-18 19:34:32.056678893 +0000
+++ config	2017-09-07 12:06:45.507029663 +0000
@@ -410,6 +410,13 @@
          case 5.1:
          case 5.2:
          case 5.3:
+         case 5.4:
+         case 6.1:
+         case 6.2:
+         case 6.3:
+         case 6.4:
+         case 7.1:
+         case 7.2:
             echo "   Good, the newest gfortrans can compile REAL*16 data type."
             breaksw
          default:
@@ -810,6 +817,13 @@
    case 5.1:
    case 5.2:
    case 5.3:
+   case 5.4:
+   case 6.1:
+   case 6.2:
+   case 6.3:
+   case 6.4:
+   case 7.1:
+   case 7.2:
       echo "   Good, the newest gfortrans can compile REAL*16 data type."
       breaksw
    default:
--- comp.orig	2017-07-19 11:55:05.490262037 +0000
+++ comp	2017-09-07 12:06:45.507029663 +0000
@@ -1593,6 +1593,13 @@
          case 5.1:
          case 5.2:
          case 5.3:
+         case 5.4:
+         case 6.1:
+         case 6.2:
+         case 6.3:
+         case 6.4:
+         case 7.1:
+         case 7.2:
             if ($MODULE == cosmo)  set OPT='-O0'
             if ($MODULE == dcscf)  set OPT='-O0'
             if ($MODULE == tddgrd) set OPT='-O0'
@@ -1823,6 +1830,13 @@
          case 5.1:
          case 5.2:
          case 5.3:
+         case 5.4:
+         case 6.1:
+         case 6.2:
+         case 6.3:
+         case 6.4:
+         case 7.1:
+         case 7.2:
             if ($MODULE == cosmo)  set OPT='-O0'  # same issue as seen in 4.6
             if ($MODULE == dcscf)  set OPT='-O0'  # exam44, continues from 4.7
             if ($MODULE == tddgrd) set OPT='-O0'  # exam41, continues from 4.6
@@ -2074,6 +2088,13 @@
          case 5.1:
          case 5.2:
          case 5.3:
+         case 5.4:
+         case 6.1:
+         case 6.2:
+         case 6.3:
+         case 6.4:
+         case 7.1:
+         case 7.2:
             if ($MODULE == cosmo)  set OPT='-O0'
             if ($MODULE == dcscf)  set OPT='-O0'
             if ($MODULE == tddgrd) set OPT='-O0'
@@ -2191,6 +2212,13 @@
          case 5.1:
          case 5.2:
          case 5.3:
+         case 5.4:
+         case 6.1:
+         case 6.2:
+         case 6.3:
+         case 6.4:
+         case 7.1:
+         case 7.2:
             if ($MODULE == cosmo)  set OPT='-O0'
             if ($MODULE == dcscf)  set OPT='-O0'
             if ($MODULE == tddgrd) set OPT='-O0'
